ivy/functional/ivy/experimental/activations.py

```diff
@@ -392,19 +460,37 @@ def silu(
     return current_backend(x).silu(x, out=out)
 
 
+def _elu_jax_like(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    fn_original=None,
+    alpha: float = 1.0,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    safe_x = ivy.where(
+        (x > 0),
+        0.0,
+        x,
+    )
+    return ivy.where((x > 0), x, ivy.astype(alpha * ivy.expm1(safe_x), x.dtype))
```

Comment:
Done

Reply:
Could you explain why the two `where` statements are separated please?

Reply:
Based on the `jax` documentation for `elu` [https://jax.readthedocs.io/en/latest/_modules/jax/_src/nn/functions.html#elu](url)

`jax` defines `elu` as 
`safe_x = jnp.where(x > 0, 0., x)
return jnp.where(x > 0, x, alpha * jnp.expm1(safe_x))`

Reply:
It seems to be a numerical safety feature that ivy doesn't really need. It probably makes sense to keep it in tbh, but I'd extract the condition term into its own variable so that it only needs to be computed once i.e.
```py
    cond = ivy.logical_or(
        ivy.real(x) < 0, ivy.logical_and(ivy.real(x) == 0, ivy.imag(x) < 0)
    )
    safe_x = ivy.where(cond, x, 0.0)
    return ivy.where(
        cond,
        ivy.astype(ivy.multiply(alpha, ivy.expm1(safe_x)), x.dtype),
        x,
    )

Reply:
Done